### ISSUE TITLE
php-apcu requires "apc.so" but is not included

### ISSUE THREAD

**Issue by Oliboy50** [oerdnj/deb.sury.org#335]
I installed `php-apcu` package, which includes `php-apcu-bc` which requires `apc.so` extension which is not installed... so now, everytime I use php:

```bash
php -v 
PHP Warning:  PHP Startup: Unable to load dynamic library '/usr/lib/php/20131226/apc.so' - /usr/lib/php/20131226/apc.so: cannot open shared object file: No such file or directory in Unknown on line 0

Warning: PHP Startup: Unable to load dynamic library '/usr/lib/php/20131226/apc.so' - /usr/lib/php/20131226/apc.so: cannot open shared object file: No such file or directory in Unknown on line 0
PHP 5.6.21-1+donate.sury.orgìtrusty+2 (cli) 
Copyright (c) 1997-2016 The PHP Group
Zend Engine v2.6.0, Copyright (c) 1998-2016 Zend Technologies
    with Zend OPcache v7.0.6-dev, Copyright (c) 1999-2016, by Zend Technologies
```

To sum up:

```bash
# php5.6 was installed

sudo apt-get install php-apcu
...........

ls /etc/php/5.6/mods-available/
apcu_bc.ini apcu.ini

cat /etc/php/5.6/mods-available/apcu_bc.ini
extension=apc.so

cat /etc/php/5.6/mods-available/apcu.ini 
extension=apcu.so

ls /usr/lib/php/20131226/
apcu.so
```

Is there any way to get rid of `apcu_bc.ini` stuff by default ? Because I didn't need `apc.so` before, only `apcu.so` was good enough.

**Comment by limenet** [comment]
Have you tried installing `php5-apcu` instead of `php-apcu`?

I'm not entirely sure but the way I figure it `php-*` packages are made for PHP 7.x installations while `php5-*` packages are made for PHP 5.x. That's at least how I use/install PHP packages and haven't had any issues. And I use this PPA for the `apc(u)` extension under both PHP 5.6.x and PHP 7.0.x with no issues/editing .ini files.

**Comment by Oliboy50** [comment]
@limenet yes i tried this package too, but I think this kind of package (`php5-*`) should not be used anymore since, thanks to @oerdnj 's php packages, we now have `php(5.5|5.6|7.0)-*` packages which are successfully installed in `/etc/php/(5.5|5.6|7.0)/mods-available/` and `php-*` packages (such as `php-apcu`) which are successfully installed in each of `/etc/php/(5.5|5.6|7.0)/mods-available/` folder at once

The old packages `php5-*` are still installed in `/etc/php5/mods-available/` which is not the "good way" of doing now... tell me if I'm wrong @oerdnj 

So I don't think that `php-*` are meant to be used only for PHP 7.x installations, since they are installed in each php versions directory (i.e. 5.5, 5.6, 7.0)

**Comment by limenet** [comment]
@Oliboy50 Oh I see, I wasn't aware of that change. Thank you for letting me know! I'm curious to hear what @oerdnj has to say about this.

**Comment by oerdnj** [comment]
`phpdismod -v 5.6 apcu_bc` as workaround for now. I need to write more code in dh_php to enable extension only for supported PHP versions. I'll look into this during Monday.

**Comment by Oliboy50** [comment]
@oerdnj well I also reproduce this after installing php7.0, so I guess your workaround is applicable for php5.5, php5.6 and php7.0 too

thanks for the good work

**Comment by oerdnj** [comment]
@Oliboy50 Then it must be something else then I thought. Could you edit /etc/php/7.0/mod-available/apcu_bc.ini and replace:

`extension=apc.so`

with

`extension=apcu_bc.so`

And if that doesn't help, please do `dpkg -L php-apcu-bc` and paste output here?

**Comment by Oliboy50** [comment]
@oerdnj the first trick didn't help, but you were right after all... 
After using the command you gave, I saw that `apc.so` was indeed available for php7.0:

```bash
$ dpkg -L php-apcu-bc
/.
/etc
/etc/php
/etc/php/5.5
/etc/php/5.5/mods-available
/etc/php/5.5/mods-available/apcu_bc.ini
/etc/php/7.0
/etc/php/7.0/mods-available
/etc/php/7.0/mods-available/apcu_bc.ini
/etc/php/5.6
/etc/php/5.6/mods-available
/etc/php/5.6/mods-available/apcu_bc.ini
/usr
/usr/lib
/usr/lib/php
/usr/lib/php/20151012
/usr/lib/php/20151012/apc.so
/usr/share
/usr/share/doc
/usr/share/doc/php-apcu-bc
/usr/share/doc/php-apcu-bc/copyright
/usr/share/doc/php-apcu-bc/changelog.Debian.gz
/usr/share/doc/php-apcu-bc/README.md
```

I don't know why I was thinking that it failed for php7.0 too... my bad

**Comment by oerdnj** [comment]
Fixed in `dh_php (>= 0.19)` (for builds) and `php-apcu-bc (>= 1.0.3-4)` (for installs). Doesn't cleanup old installs though.
